File: Server_DB/Server_spring/digitaltwin-backend-springboot-main/src/main/java/com/fifthdimension/digital_twin/config/SwaggerConfig.java
package com.fifthdimension.digital_twin.config;

import io.swagger.v3.oas.annotations.OpenAPIDefinition;
import io.swagger.v3.oas.annotations.info.Info;
import io.swagger.v3.oas.models.Components;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.security.SecurityRequirement;
import io.swagger.v3.oas.models.security.SecurityScheme;
import io.swagger.v3.oas.models.servers.Server;
import jakarta.annotation.PostConstruct;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Arrays;

@OpenAPIDefinition(
        info = @Info(
                title = "Fifth Dimension Digital Twin API",
                description = "Digital Twin Spring Boot 프로젝트 API 명세"
        )
)
@Configuration
public class SwaggerConfig {

    @Value("${server.servlet.context-path}")
    private String base_path;

    @Value("${server.host}")
    private String serverHost;

    private Server[] servers;

    @PostConstruct
    public void initializeServers() {
        servers = new Server[]{
                new Server().url(serverHost + base_path).description("Prod"),
                new Server().url("http://localhost:8080" + base_path).description("Dev")
        };
    }

    // 추후 JWT 토큰 사용할 부분 고려해서 적용
    @Bean
    public OpenAPI customOpenAPI() {
        return new OpenAPI()
                .servers(Arrays.stream(servers).toList())
                .components(new Components()
                        .addSecuritySchemes("JWT-Token", new SecurityScheme()
                                .type(SecurityScheme.Type.HTTP)
                                .scheme("bearer")
                                .bearerFormat("JWT")
                                .in(SecurityScheme.In.HEADER)
                                .name("Authorization")))
                .addSecurityItem(new SecurityRequirement().addList("JWT-Token"));
    }

}

File: Server_DB/Server_express/digitaltwin-backend-expressjs-main/src/config/swaggerConfig.js
const swaggerJSDoc = require('swagger-jsdoc');
const path = require('path');
const { host, basePath } = require('../config/env');

const options = {
  definition: {
    openapi: '3.0.0',
    info: {
      title: 'Fifth Dimension Digital Twin API',
      version: '1.0.0',
      description: 'Digital Twin Express 프로젝트 API Docs',
    },
    servers: [
      {
        url: `${host}${basePath}`,
        description: 'Prod',
      },
      {
        url: `http://localhost:3000${basePath}`,
        description: 'Dev',
      },
    ],
    components: {
      securitySchemes: {
        'JWT-Token': {
          type: 'http',
          scheme: 'bearer',
          bearerFormat: 'JWT',
          name: 'Authorization',
          in: 'header',
          description: 'JWT Authorization header using the Bearer scheme. Example: "Authorization: Bearer {token}"'
        }
      }
    },
    security: [
      {
        'JWT-Token': []
      }
    ]
  },
  apis: [path.join(__dirname, '../presentation/*.js')],
};

const specs = swaggerJSDoc(options);
module.exports = specs;